### PR TITLE
Support full dropdown functionality

### LINKED ISSUE #4: Support all dropdown types
http://semantic-ui.com/modules/dropdown.html

### LINKED ISSUE #149: Dropdown fails to open on initial render with dynamic className change
I was attempting to add a loading indicator to the `Dropdown` component by dynamically adding a `loading` class using [`classNames`](https://www.npmjs.com/package/classnames) like so:

``` jsx
<Dropdown
  className={cx({loading: isDropdownLoading}, 'fluid search option')}
  name='clientId'
  options={dropdownOptions}
  defaultValue={client.id}
/>
```

Which resulted in the dropdown failing to open once the page rendered (only the first time the component rendered, not the second).

Reproduce with Stardust `v0.9.6` using semantic.js `v2.1.6`. Note the scope of this may apply more broadly to changes in props per @levithomason

### PR DESCRIPTION
**Pull request by levithomason**
### [Try It](http://deweybot.taplatform.net/qa/src/stardust/feature/dynamic-dropdown-options/#Dropdown-Types-Dropdown)

There are **3 examples** written.  More will be added in other PRs later.
### Not as large as it looks

This PR includes 3 primary things:
1. Doc Site Examples
2. Test updates
3. Dropdown rewrite
### Done

Fixes #4 supporting all dropdown permutations.  Also fixes #149 supporting dynamic options.
- [x] add dropdown menu, item, and divider sub components
- [x] support icons
- [x] support item text/value/descriptions
- [x] add extended dropdown example using dropdown subcomponents
- [x] handle regular dropdowns (menu only style)
- [x] handle selection dropdowns, with hidden input for values
- [x] separate component / plugin prop types
- [x] componentDidUpdate
- [x] add async options example

### CODE REVIEW

File: docs/app/Examples/modules/Dropdown/Content/AsyncOptions.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  render() {
+    const { options, isFetching } = this.state
+    return (
+      <div>
+        <Button onClick={this.updateItems}>
+          Fetch Items
+        </Button>
+        <Dropdown
+          className={`search selection multiple ${isFetching && 'disabled loading'}`}
```
Comment (ghost) [inline docs/app/Examples/modules/Dropdown/Content/AsyncOptions.js]: I had opened an issue related to programmatic use of the loading indicator with https://github.com/TechnologyAdvice/stardust/issues/149. Can that be closed with this PR?

File: src/modules/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -1,111 +1,97 @@
...
-      onRemove: this.props.onRemove,
-    })
+    this.refresh()
   }
 
   componentDidUpdate(prevProps, prevState) {
-    this.element.dropdown('refresh')
+    if (prevProps === this.props) {
+      this.refresh()
+    }
```
Comment (ghost) [inline src/modules/Dropdown/Dropdown.js]: When the component updates, wouldn't we want to refresh only if `props` did change? And, if so, wouldn't we do that with a shallow or deep compare given the `options` and `children` props are objects themselves?

**Comment by ghost** [comment]
This is great! I left a couple of comments. Other than that the code looks great. Thanks for updating Examples given we have a new way to structure our menus now (e.g. `<Dropdown.Menu />`).

Ship when ready: :star:

File: docs/app/Examples/modules/Dropdown/Types/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+import { Dropdown } from 'stardust'
+
+export default class DropdownDropdownExample extends Component {
+  render() {
+    return (
+      <Dropdown text='File' action='hide'>
```
Comment (levithomason) [inline docs/app/Examples/modules/Dropdown/Types/Dropdown.js:7]: If you only want to show a menu, and not have the text update, set the correct [Semantic UI action](http://semantic-ui.com/modules/dropdown.html#specifying-select-action).  In this case, we just hide the dropdown on select.  This leaves the `File` text in the dropdown space so it looks and works like a normal dropdown menu (not a selection dropdown).

File: docs/app/Examples/modules/Dropdown/States/Disabled.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react'
+import { Dropdown } from 'stardust'
+
+export default class DropdownDisabledExample extends Component {
+  render() {
+    return (
+      <Dropdown className='disabled' text='Dropdown'>
+        <Dropdown.Menu>
+          <Dropdown.Item text='Choice 1' />
```
Comment (tslater) [inline docs/app/Examples/modules/Dropdown/States/Disabled.js:9]: I can't get this example to work. the Dropdown.Items don't show up.

Comment (levithomason) [inline docs/app/Examples/modules/Dropdown/States/Disabled.js:9]: Linking #346

**Pull request merged**